compiler/rustc_typeck/src/check/generator_interior.rs

```diff
@@ -387,6 +387,18 @@ impl<'a, 'tcx> Visitor<'tcx> for InteriorVisitor<'a, 'tcx> {
             ty.needs_drop(self.fcx.tcx, self.fcx.param_env)
         };
 
+        let find_parent_expr = |mut hir_id| {
+            let hir = self.fcx.tcx.hir();
+            hir_id = hir.find_parent_node(hir_id)?;
+            loop {
```

Comment:
Can you use https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/hir/map/struct.Map.html#method.parent_iter here?

Reply:
Ah, thanks! I didn't know about that method. I think that's way cleaner, so I rewrote my code to use that. I also added a doc comment to `find_parent_node` to point people towards `parent_iter` when more appropriate.